### ISSUE TITLE
case sensitive bundle id failing

### ISSUE THREAD

**Issue by trops** [fastlane/fastlane#8612]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [ ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I am using match to generate the profiles and certificates and I have existing bundle id with a capital letter for the app name. 

The provisioning profiles are being created for the LOWERCASE bundle id I also have in my developer portal. The fast lane build fails for code signing. I have downloaded the generated profile from github repo where it was uploaded by fast lane and the error is that the profile doesn't match the bundle id (says it built with lowercase letter when I explicitly specified uppercase).

I tried editing the profile in the developer portal to use the proper uppercase bundle id (as mentioned I have one lowercase and one uppercase) and it fails as well. 

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[INSERT OUTPUT HERE]

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                        |
| --------------------------- | -------------------------------------------- |
| OS                          | 10.12.3                                      |
| Ruby                        | 2.4.0                                        |
| Bundler?                    | false                                        |
| Git                         | git version 2.9.3                            |
| Installation Source         | ~/.rvm/rubies/ruby-2.4.0/bin/fastlane        |
| Host                        | Mac OS X 10.12.3 (16D32)                     |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.0/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2j  26 Sep 2016                  |
| Is contained                | false                                        |
| Is homebrew                 | false                                        |
| Is installed via Fabric.app | false                                        |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer// |
| Xcode Version               | 8.2.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.22.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    # we want to fetch the identifiers so that we can fetch them 
    # automatically
    team_id = CredentialsManager::AppfileConfig.try_fetch_value(:team_id)
    app_identifier = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)
    profile = ENV["sigh_#{app_identifier}_appstore"]
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do

    team_id = "KU7VY2RK28" #CredentialsManager::AppfileConfig.try_fetch_value(:team_id)
    app_identifier = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)
    profile = "sigh_com.databake.Swingbot_development" #ENV["sigh_#{app_identifier}_development"]
   
    match(type: "development", git_url: "https://github.com/trops/Swingbot-match-certs.git")
    gym(scheme: 'Swingbot', xcargs: "PROVISIONING_PROFILE_SPECIFIER='#{profile}' DEVELOPMENT_TEAM='#{team_id}'")
    pilot

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do

    team_id = "KU7VY2RK28"
    app_identifier = CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier)
    profile = ENV["sigh_#{app_identifier}_appstore"]
    
    match(type: "appstore", git_url: "https://github.com/trops/Swingbot-match-certs.git")
    # snapshot
    gym(scheme: 'Swingbot', xcargs: "PROVISIONING_PROFILE_SPECIFIER='#{profile}' DEVELOPMENT_TEAM='#{team_id}'")
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.databake.Swingbot" # The bundle identifier of your app
apple_id "john@databake.com" # Your Apple email address

team_id "KU7VY2RK28"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.22.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.14.6       |
| io-console                | 0.4.6        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| excon                     | 0.55.0       |
| faraday                   | 0.11.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170223 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| json                      | 2.0.2        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| faraday_middleware        | 0.11.0.1     |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.10.1       |
| activesupport             | 4.2.8        |
| openssl                   | 2.0.2        |
</details>

*generated on:* **2017-03-22**
</details>

**Comment by jcharr1** [comment]
I have a very similar problem. We have an app in the App Store called Visual Dictionary. The bundle id ends in .visualdictionary. For some reason, we also have an app id registered with Apple that ends in .VisualDictionary. I think this was an older version that's since been deleted from the App Store, but we can't delete it because the dev portal says it's still in use in the App Store. When I try to use match for the bundle id that's all in lower case, it always creates profiles for the mixed case id that's no longer in the store. I've been able to work around this somewhat by created the profiles with match, going into the dev portal, manually changing the bundle id associated with the profile to the correct one, and then manually downloading the updated profile. It really kind of hurts match's usefulness for this particular app. It would be really nice if there were an option to force it to match the case of the bundle id you provide to match. Or at least have the option to choose when there are profiles with the same name but with different cases.

I also had a similar problem with pilot with the same app. It kept trying to upload the binary to the old mixed case version and failing even though I told pilot to use the all lowercase version. I was able to workaround this by also specifying the Apple ID.
